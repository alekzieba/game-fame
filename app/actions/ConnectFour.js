import { firebaseapp } from '../constants/firebase';

export const CLICK_COLUMN = 'CLICK_COLUMN';
export const SET_BOARD = 'SET_BOARD';
export const CREATE_BOARD = 'CREATE_BOARD';
export const GET_BOARD = 'CREATE_BOARD';
export const UPDATE_BOARD = 'UPDATE_BOARD';

export function checkVertical(board) {
  return board.reduce((winner, col) => {
    let currentWinner = '';

    // loop through first three circles
    for (let i = 0; i < 3; i += 1) {
      if (
        col[i] !== '' &&
        col[i] === col[i + 1] &&
        col[i] === col[i + 2] &&
        col[i] === col[i + 3]
      ) {
        currentWinner = col[i];
      } else {
        currentWinner = winner;
      }
    }

    return currentWinner;
  }, '');
}

export function checkHorizontal(board) {
  for (let i = 0; i < 4; i += 1) {
    // loop through all circles
    const currentWinner = board[i].reduce((winner, circle, index) => {
      if (
        circle !== '' &&
        circle === board[i + 1][index] &&
        circle === board[i + 2][index] &&
        circle === board[i + 3][index]
      ) {
        return circle;
      }
      return winner;
    }, '');

    // if a winner is found exit function returning winner
    // otherwise continue outer for loop
    if (currentWinner) return currentWinner;
  }

  return '';
}

export function checkDiagonal1(board) {
  // within board loop through first 4 columns
  for (let i = 0; i < 4; i += 1) {
    // within each column loop through first 3 circles
    for (let j = 0; j < 3; j += 1) {
      if (
        board[i][j] !== '' &&
        board[i][j] === board[i + 1][j + 1] &&
        board[i][j] === board[i + 2][j + 2] &&
        board[i][j] === board[i + 3][j + 3]
      ) {
        return board[i][j];
      }
    }
  }

  return '';
}

export function checkDiagonal2(board) {
  // within board loop through last 4 columns
  for (let i = 3; i < 7; i += 1) {
    // within each column loop through first 3 circles
    for (let j = 0; j < 3; j += 1) {
      if (
        board[i][j] !== '' &&
        board[i][j] === board[i - 1][j + 1] &&
        board[i][j] === board[i - 2][j + 2] &&
        board[i][j] === board[i - 3][j + 3]
      ) {
        return board[i][j];
      }
    }
  }

  return '';
}

export function calculateWinner(board) {
  return (
    checkVertical(board) ||
    checkHorizontal(board) ||
    checkDiagonal1(board) ||
    checkDiagonal2(board)
  );
}

export function clickColumn(
  colIndex,
  board,
  player1IsNext,
  player1,
  player2,
  gameIsWon,
  gameKey
) {
  // if available circles in column
  if (board[colIndex][0] === '') {
    // add current player to last empty string in column
    const copyBoard = board.slice().map((column, index) => {
      if (index === colIndex) {
        const colClone = column.slice(0);
        const lastIndex = colClone.lastIndexOf('');

        colClone[lastIndex] = player1IsNext ? 'player1' : 'player2';

        return colClone;
      }
      return column;
    });

    // A player has won the game
    if (calculateWinner(copyBoard)) {
      const updates = {};
      updates[`${gameKey}/board`] = copyBoard;
      updates[`${gameKey}/gameIsWon`] = true;
      firebaseapp
        .database()
        .ref()
        .update(updates);

      return {
        type: CLICK_COLUMN,
        payload: {
          board: copyBoard,
          gameIsWon: true,
          player1,
          player2,
          player1IsNext,
          gameKey
        }
      };
    }

    // No player has won
    const updates = {};
    updates[`${gameKey}/board`] = copyBoard;
    updates[`${gameKey}/player1IsNext`] = !player1IsNext;
    firebaseapp
      .database()
      .ref()
      .update(updates);

    return {
      type: CLICK_COLUMN,
      payload: {
        board: copyBoard,
        gameIsWon,
        player1,
        player2,
        player1IsNext: !player1IsNext,
        gameKey
      }
    };
  }
  return null;
}

export function setPost(gameData) {
  return gameData;
}

export function getBoard(gameKey, user1, user2) {
  // eslint-disable-next-line func-names
  return function(dispatch) {
    const dbdRef = firebaseapp.database().ref(gameKey);

    let newBoard = [];
    let newGameIsWon = false;
    let newPlayer1IsNext = false;

    dbdRef
      .once('value', snapshot => {
        newBoard = (snapshot.val() && snapshot.val().board) || [];
        newGameIsWon = (snapshot.val() && snapshot.val().gameIsWon) || false;
        newPlayer1IsNext =
          (snapshot.val() && snapshot.val().player1IsNext) || false;
      })
      .then(() =>
        dispatch(
          setPost({
            type: SET_BOARD,
            payload: {
              player1IsNext: newPlayer1IsNext,
              player1: user1,
              player2: user2,
              gameIsWon: newGameIsWon,
              gameKey,
              board: newBoard
            }
          })
        )
      )
      .catch();
  };
}

export const dbChanged = (gameKey, user1, user2, dispatch) => {
  firebaseapp
    .database()
    .ref(gameKey)
    .on('child_changed', data => {
      if (data.val()) {
        dispatch(getBoard(gameKey, user1, user2));
      }
    });
};

export function createBoard(newGameKey, user1, user2) {
  // eslint-disable-next-line no-console
  console.log('LOG: Create Board');
  const initialState = {
    player1IsNext: true,
    player1: user1,
    player2: user2,
    gameIsWon: false,
    gameKey: newGameKey,
    board: [
      ['', '', '', '', '', ''],
      ['', '', '', '', '', ''],
      ['', '', '', '', '', ''],
      ['', '', '', '', '', ''],
      ['', '', '', '', '', ''],
      ['', '', '', '', '', ''],
      ['', '', '', '', '', '']
    ]
  };

  firebaseapp
    .database()
    .ref(`${initialState.gameKey}/board`)
    .set(initialState.board);
  firebaseapp
    .database()
    .ref(`${initialState.gameKey}/player1IsNext`)
    .set(initialState.player1IsNext);
  firebaseapp
    .database()
    .ref(`${initialState.gameKey}/gameIsWon`)
    .set(initialState.gameIsWon);
  firebaseapp
    .database()
    .ref(`${initialState.gameKey}/player1`)
    .set(initialState.player1);
  firebaseapp
    .database()
    .ref(`${initialState.gameKey}/player2`)
    .set(initialState.player2);
  firebaseapp
    .database()
    .ref(`${initialState.gameKey}/gameKey`)
    .set(initialState.gameKey);

  return {
    type: CREATE_BOARD,
    payload: initialState
  };
}
