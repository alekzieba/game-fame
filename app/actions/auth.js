// @flow
// import * as firebase from 'firebase';
import { ipcRenderer } from 'electron';
import { google } from 'googleapis';
import { firebaseapp } from '../constants/firebase';

export const USER_SIGNED_IN = 'USER_SIGNED_IN';
export const USER_SIGNED_OUT = 'USER_SIGNED_OUT';

const usersRef = firebaseapp.database().ref('users');

export function promptUserSignIn(loginHandler) {
  return (dispatch: (action: actionType) => void) => {
    ipcRenderer.removeAllListeners('signin:google');
    ipcRenderer.send('signin:google');
    ipcRenderer.once('signin:google', (event, googleResponse) => {
      // Get user's profile information
      const oauth2Client = new google.auth.OAuth2();
      oauth2Client.setCredentials(googleResponse.tokens);
      const oauth2 = google.oauth2({
        auth: oauth2Client,
        version: 'v2'
      });
      oauth2.userinfo.get((err, res) => {
        if (err) {
          console.log(err);
        } else {
          // https://gist.github.com/anantn/4323967
          const sanitizedEmail = res.data.email
            .replace(/@/g, '|')
            .replace(/\./g, '=');
          console.log(sanitizedEmail);
          usersRef.child(sanitizedEmail).transaction(
            currentUserData => {
              if (currentUserData === null) {
                dispatch({
                  type: USER_SIGNED_IN,
                  payload: res
                });
                loginHandler(true);
                return {
                  email: res.data.email,
                  name: res.data.name,
                  game_ids: []
                };
              }
            },
            () => {
              // params: error, committed
              // do nothing
            }
          );
        }
      });
    });
  };
}

export function userSignOut() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({
      type: USER_SIGNED_OUT
    });
  };
}
