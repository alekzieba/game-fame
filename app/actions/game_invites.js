import _ from 'lodash';

import type { Dispatch } from '../reducers/types';
import { firebaseapp } from '../constants/firebase';

export const GET_GAME_INVITE = 'GET_GAME_INVITE';
export const REMOVE_GAME_INVITE = 'REMOVE_GAME_INVITE';
export const CLEAR_GAME_INVITES = 'CLEAR_GAME_INVITES';

export function getGameInvites(sanitizedEmail) {
  return (dispatch: Dispatch) => {
    const gameIds = firebaseapp
      .database()
      .ref('users')
      .child(sanitizedEmail)
      .child('game_invite_ids');
    gameIds.on('value', gameIdsSnapshot => {
      // for each game ID, obtain the game object
      dispatch({
        type: CLEAR_GAME_INVITES
      });
      _.each(gameIdsSnapshot.val(), gameId => {
        if (gameId !== undefined) {
          firebaseapp
            .database()
            .ref('game_info')
            .child(gameId)
            .on('value', snapshot => {
              dispatch({
                type: GET_GAME_INVITE,
                payload: snapshot.val()
              });
            });
        }
      });
    });
  };
}

export function createGameInvite(
  {
    type,
    lastMoveTime,
    whoseTurn,
    status,
    inviterEmail,
    inviterName,
    inviterAvatar,
    invitedEmail
  },
  gameId,
  inviterFriendsList,
  inviterGames,
  callback
) {
  return async () => {
    // Add a reference to this object in the invited user's object
    const friendRef = firebaseapp.database().ref(`users/${invitedEmail}`);
    friendRef.once('value', snapshot => {
      if (snapshot.val()) {
        // Create an object that stores the game's info
        const gameInfoRef = firebaseapp.database().ref(`game_info/${gameId}`);
        const toSet = {
          game_id: gameId,
          type,
          last_move_time: lastMoveTime.toString(),
          whose_turn: whoseTurn,
          status,
          inviter_email: inviterEmail,
          inviter_name: inviterName,
          inviter_avatar: inviterAvatar,
          invited_email: invitedEmail,
          invited_avatar: snapshot.val().picture
        };
        console.log(toSet);
        gameInfoRef.set(toSet); // the listener made for game invites should automatically trigger

        const friendGameInvites = friendRef.child('game_invite_ids');
        friendGameInvites.once('value', invitesIdsSnapshot => {
          const currInviteIds = invitesIdsSnapshot.val();
          if (currInviteIds) {
            currInviteIds.push(gameId);
            friendGameInvites.update(currInviteIds);
          } else {
            friendGameInvites.set([gameId]);
          }
          callback(true);
        });

        const userGamesRef = firebaseapp
          .database()
          .ref(`users/${inviterEmail}/game_ids`);
        userGamesRef.once('value', gameIdsSnapshot => {
          const currInviteIds = gameIdsSnapshot.val();
          if (currInviteIds) {
            currInviteIds.push(gameId);
            userGamesRef.update(currInviteIds);
          } else {
            userGamesRef.set([gameId]);
          }
          callback(true);
        });

        // Add to friend's list
        const userFriendsListRef = firebaseapp
          .database()
          .ref(`users/${inviterEmail}/friends_list`);
        const friendInfo = {
          picture: snapshot.val().picture,
          name: snapshot.val().name,
          email: snapshot.val().email
        };
        if (inviterFriendsList === undefined) {
          userFriendsListRef.set([friendInfo]);
        } else {
          // make sure we don't add duplicates
          const newInviterFriendsList = _.filter(
            inviterFriendsList,
            o => o.email !== snapshot.val().email
          );
          newInviterFriendsList.push(friendInfo);
          userFriendsListRef.set(newInviterFriendsList);
        }
      } else {
        callback(false);
      }
    });
  };
}

export function acceptGameInvite(
  gameId,
  gameInviteIds,
  gameIds,
  sanitizedEmail
) {
  return async () => {
    const fixedInvitesArr = _.without(gameInviteIds, gameId);
    firebaseapp
      .database()
      .ref(`users/${sanitizedEmail}/game_invite_ids`)
      .set(fixedInvitesArr);

    if (gameIds === undefined) {
      firebaseapp
        .database()
        .ref(`users/${sanitizedEmail}/game_ids`)
        .set([gameId]);
    } else {
      gameIds.push(gameId);
      firebaseapp
        .database()
        .ref(`users/${sanitizedEmail}/game_ids`)
        .set(gameIds);
    }
  };
}

export function declineGameInvite(gameId, gameInviteIds, sanitizedEmail) {
  return () => {
    const fixedArr = _.without(gameInviteIds, gameId);
    firebaseapp
      .database()
      .ref(`users/${sanitizedEmail}/game_invite_ids`)
      .set(fixedArr);
  };
}

// export function acceptGameInvite(gameInviteId) {
//   // Move the id into the game_info_ids prop and remove from game_invite_ids
//   return () => {
//
//   }
// }
