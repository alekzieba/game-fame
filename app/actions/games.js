// @flow
import type { Dispatch } from '../reducers/types';
import { firebaseapp } from '../constants/firebase';

export const GET_GAMES = 'GET_GAMES';
export const CREATE_GAME = 'CREATE_GAME';

let gamesRef;

export function getGames(userId) {
  return (dispatch: Dispatch) => {
    gamesRef = firebaseapp.database().ref(`games/${userId}`);
    gamesRef.on('value', snapshot => {
      dispatch({
        type: GET_GAMES,
        payload: snapshot.val()
      });
    });
  };
}

// when creating a game, only succeed if the partner also
//   has an account so that the IDs can be stored properly
