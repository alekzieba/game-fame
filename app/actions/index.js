import * as firebase from 'firebase';

const config = {
  apiKey: 'AIzaSyC-5n96g7DgnjdAZXxQedr8kbeLvgBLXgA',
  authDomain: 'tic-tac-toe-18e8c.firebaseapp.com',
  databaseURL: 'https://tic-tac-toe-18e8c.firebaseio.com',
  projectId: 'tic-tac-toe-18e8c',
  storageBucket: 'tic-tac-toe-18e8c.appspot.com',
  messagingSenderId: '781435447234'
};
const firebaseapp = firebase.initializeApp(config);

function convertBoard(board, type1, type2) {
  const boardRet = board.slice();
  for (let i = 0; i < board.length; i += 1) {
    if (board[i] === type1) {
      boardRet[i] = type2;
    }
  }
  return board;
}

export function fillSquare(index, board, symbol, gameKey) {
  let newBoard = board.slice();
  newBoard = convertBoard(newBoard, null, 'null');
  newBoard[index] = symbol;
  const xIsTrue = !(symbol === 'X');
  firebaseapp
    .database()
    .ref(`TicTacToe/games/${gameKey}/0`)
    .set(newBoard);
  firebaseapp
    .database()
    .ref(`TicTacToe/games/${gameKey}/1`)
    .set(xIsTrue);
  newBoard = convertBoard(newBoard, 'null', null);
  return {
    type: 'FILL_SQUARE',
    payload: {
      board: newBoard,
      xIsTrue
    }
  };
}

export function resetBoard(gameKey) {
  let newBoard = Array(9).fill('null');
  firebaseapp
    .database()
    .ref(`TicTacToe/games/${gameKey}/0`)
    .set(newBoard);
  firebaseapp
    .database()
    .ref(`TicTacToe/games/${gameKey}/1`)
    .set(true);
  newBoard = convertBoard(newBoard, 'null', null);
  return {
    type: 'RESET_BOARD',
    payload: {
      board: newBoard,
      xIsTrue: true
    }
  };
}

export function getBoard(gameKey) {
  const gameBoardRef = firebaseapp.database().ref(`TicTacToe/games/${gameKey}`);

  return (dispatch: Dispatch) => {
    gameBoardRef.on('value', snapshot => {
      dispatch({
        type: 'GET_BOARD',
        payload: {
          board: snapshot.val(),
          gameKey
        }
      });
    });
  };
}

export function createBoard(gameKey) {
  const boardRef = firebaseapp.database().ref('TicTacToe');
  const board = Array(9).fill('null');
  const gameElem = boardRef.child(`games/${gameKey}`);

  gameElem.set([board, true]);
  return {
    type: 'NEW_BOARD',
    payload: {
      board,
      xIsTrue: true,
      gameKey
    }
  };
}
