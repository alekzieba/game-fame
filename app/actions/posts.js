// @flow
import type { Dispatch } from '../reducers/types';
import { firebaseapp } from '../constants/firebase';

export const GET_POSTS = 'GET_POSTS';

const postsRef = firebaseapp.database().ref('posts');

export function getPosts() {
  return (dispatch: Dispatch) => {
    postsRef.on('value', snapshot => {
      dispatch({
        type: GET_POSTS,
        payload: snapshot.val()
      });
    });
  };
}

export function createPost(post) {
  return () => {
    const newPostRef = postsRef.push();
    newPostRef.set(post);
  };
}

export function deletePost(key) {
  return () => {
    postsRef.child(key).remove();
  };
}
