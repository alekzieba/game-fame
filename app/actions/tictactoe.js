import { firebaseapp } from '../constants/firebase';

function convertBoard(board, type1, type2) {
  const boardRet = board.slice();
  for (let i = 0; i < board.length; i += 1) {
    if (board[i] === type1) {
      boardRet[i] = type2;
    }
  }
  return boardRet;
}

export function fillSquare(index, board, symbol, gameKey, userMakingMove) {
  let firstUser = null;

  //  dbdRef
  //      .once('value', snapshot => {
  //        newBoard = (snapshot.val() && snapshot.val().board) || [];
  //        newGameIsWon = (snapshot.val() && snapshot.val().gameIsWon) || false;
  //        newPlayer1IsNext =
  //          (snapshot.val() && snapshot.val().player1IsNext) || false;
  //      })
  //      .then(() =>
  //        dispatch(
  //          setPost({
  //            type: SET_BOARD,
  //            payload: {
  //              player1IsNext: newPlayer1IsNext,
  //              player1: user1,
  //              player2: user2,
  //              gameIsWon: newGameIsWon,
  //              gameKey,
  //              board: newBoard
  //            }
  //          })
  //        )
  //      )
  //      .catch();

  firebaseapp
    .database()
    .ref(`games/${gameKey}/3`)
    .once('value', snapshot => {
      firstUser = snapshot.val();
    });
  let newBoard = board.slice();
  newBoard = convertBoard(newBoard, null, 'null');
  newBoard[index] = symbol;
  const xIsTrue = !(symbol === 'X');
  console.log(firstUser, userMakingMove, xIsTrue);
  if (firstUser === userMakingMove && xIsTrue) {
    return {
      type: 'IGNORE'
    };
  }
  if (firstUser !== userMakingMove && !xIsTrue) {
    return {
      type: 'IGNORE'
    };
  }
  firebaseapp
    .database()
    .ref(`games/${gameKey}/0`)
    .set(newBoard);
  firebaseapp
    .database()
    .ref(`games/${gameKey}/1`)
    .set(xIsTrue);
  newBoard = convertBoard(newBoard, 'null', null);
  return {
    type: 'FILL_SQUARE',
    payload: {
      board: newBoard,
      xIsTrue
    }
  };
}

export function resetBoard(gameKey) {
  let newBoard = Array(9).fill('null');
  firebaseapp
    .database()
    .ref(`games/${gameKey}/0`)
    .set(newBoard);
  firebaseapp
    .database()
    .ref(`games/${gameKey}/1`)
    .set(true);
  newBoard = convertBoard(newBoard, 'null', null);
  return {
    type: 'RESET_BOARD',
    payload: {
      board: newBoard,
      xIsTrue: true
    }
  };
}

export function updateWinsAndLosses(gameKey, winningSymbol, firstUserEmail, secondUserEmail){
//    const gameBoardRef = firebaseapp.database().ref(`games/${gameKey}`);
  const userRef = firebaseapp.database().ref(`users`);
  let winningEmail = null;
  let losingEmail = null;
  let winningNum = 0;
  let losingNum = 0;
  if(winningSymbol === 'X'){
    winningEmail = firstUserEmail;
    losingEmail = secondUserEmail;
  }
  else{
    winningEmail = secondUserEmail;
    losingEmail = firstUserEmail;
  }
  userRef.child(winningEmail).child("wins").once('value', snapshot => {
    winningNum = snapshot.val();
  }).then(() =>{
    console.log(winningNum);
    winningNum += 1;
    userRef.child(winningEmail).child("wins").set(winningNum);
  });
  
  userRef.child(losingEmail).child("losses").once('value', snapshot => {
    losingNum = snapshot.val();
  }).then(() =>{
    console.log(losingNum);
    losingNum += 1;
    userRef.child(losingEmail).child("losses").set(losingNum);
  });
  
  return {
    type: 'UPDATE_WIN_LOSS',
    payload: {
      
    }
  };
}

export function getBoard(gameKey) {
  const gameBoardRef = firebaseapp.database().ref(`games/${gameKey}`);
  return (dispatch: Dispatch) => {
    gameBoardRef.on('value', snapshot => {
      dispatch({
        type: 'GET_BOARD',
        payload: {
          board: snapshot.val(),
          gameKey
        }
      });
    });
  };
}

export function getMessages(gameKey) {
  const gameBoardRef = firebaseapp.database().ref(`games/${gameKey}`);
  return (dispatch: Dispatch) => {
    gameBoardRef.on('value', snapshot => {
      dispatch({
        type: 'GET_MESSAGES',
        payload: {
          board: snapshot.val(),
          gameKey
        }
      });
    });
  };
}

export function createBoard(gameKey, firstUser = '') {
  const gameElem = firebaseapp.database().ref(`games/${gameKey}`);
  const board = Array(9).fill('null');
  const message = [
    'ADMIN: Welcome to the chat!',
    "ADMIN: Type anything here and click 'Send' to submit."
  ];
  console.log("Started")
  gameElem.set([board, true, message, firstUser]);
  console.log("Ended")
  return {
    type: 'NEW_BOARD',
    payload: {
      board,
      xIsTrue: true,
      gameKey,
      message
    }
  };
}

export function submitMessage(gameKey, message, messageList) {
  const newMessageList = messageList.slice();
  newMessageList.push(message);
  firebaseapp
    .database()
    .ref(`games/${gameKey}/2`)
    .set(newMessageList);

  return {
    type: 'SUBMIT_MESSAGE',
    payload: {
      messageList: newMessageList
    }
  };
}
