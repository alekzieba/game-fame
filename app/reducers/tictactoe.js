/* eslint-disable */
// eslint says it wants array destructuring
//   but I'm too lazy to Google it right now

export default function(
  state = { board: Array(9).fill(null), gameKey: '', xIsTrue: true },
  action
) {
  let board = Array(9).fill(null);
  switch (action.type) {
    case 'FILL_SQUARE':
      return {
        ...state,
        xIsTrue: action.payload.xIsTrue,
        board: action.payload.board
      };
    case 'RESET_BOARD':
      return {
        ...state,
        xIsTrue: action.payload.xIsTrue,
        board: Array(9).fill(null)
      };
    case 'GET_BOARD':
      board = action.payload.board[0].slice();
      for (let i = 0; i < board.length; i += 1) {
        if (board[i] === 'null') {
          board[i] = null;
        }
      }
      return {
        ...state,
        xIsTrue: action.payload.board[1],
        gameKey: action.payload.gameKey,
        board
      };
    case 'NEW_BOARD':
      board = action.payload.board.slice();
      for (let j = 0; j < board.length; j += 1) {
        if (board[j] === 'null') {
          board[j] = null;
        }
      }
      return {
        ...state,
        gameKey: action.payload.gameKey,
        xIsTrue: action.payload.xIsTrue,
        board
      };
    default:
      return state;
  }
}
