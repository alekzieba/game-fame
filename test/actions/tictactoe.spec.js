import { spy } from 'sinon';
import * as actions from '../../app/actions/tictactoe';

describe('actions', () => {
  it('should resetBoard should succeed', () => {
    expect(actions.resetBoard('test')).toMatchSnapshot();
  });

  // TODO write a test where getBoard succeeds

  it('should getBoard should fail for bad gameKey', () => {
    const fn = actions.getBoard('test');
    const dispatch = spy();

    fn(dispatch);
    expect(dispatch.called).toBe(false);
  });

  it('should createBoard should succeed', () => {
    expect(actions.createBoard('test')).toMatchSnapshot();
  });
});
