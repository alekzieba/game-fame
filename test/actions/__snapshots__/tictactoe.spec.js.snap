// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should resetBoard should succeed 1`] = `
Object {
  "payload": Object {
    "board": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
    "xIsTrue": true,
  },
  "type": "RESET_BOARD",
}
`;
